### PR TITLE
Renaming load_model_config() and load_config() 

### LINKED ISSUE #218: Unsuitable function names for loading configs
config: `load_model_config()` and `load_config()` are misleading names. 

`load_model_config` : contains model *and* run parameters, e.g. related to learning rate, #epochs etc. So an appropriate name would be load_run_config. It's a good idea to separate the config into a model config and training config but it's not there yet.

`load_config` : this loads and merges the various configs; maybe a good name would be `load_merge_configs`. The function is also missing documentation. In `load_config`, epoch and overwrite_config should also have default arguments `None`.

CC: @grassesi , @tjhunter

### PR DESCRIPTION
**Pull request by rushchanskii**
## Description

Renamed load_model_config() to load_run_config() and load_config() to load_merge_configs() for better semantics and clarity. Renamed all subsequent calls.
Added default values of parameters and some extra descriptions.

## Issue Number

Closes #218

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [x] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [x] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: packages/common/src/weathergen/common/config.py [hunk 1]
```diff
@@ -68,10 +68,20 @@ def save(config: Config, mini_epoch: int | None):
...
     Load a configuration file from a given run_id and mini_epoch.
     If run_id is a full path, loads it from the full path.
+    
+    Args:
+        run_id: Run ID of the pretrained WeatherGenerator model
+        mini_epoch: Mini-epoch of the checkpoint to load. -1 indicates last checkpoint available.
+        model_path: Path to the model directory. If None, uses the model_path from private config.
+        
+    Returns:
+        Configuration object loaded from the specified run and mini_epoch.
```
Comment (grassesi) [inline packages/common/src/weathergen/common/config.py]: Thanks for the documentation :+1:
